modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,37 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W_SIMPLE Model : public Net
+     {
+     public:
+         Model(const std::string& model, const std::string& config = "");
+         Model(const Net& network);
+
+         Model& setInputSize(const Size& size);
+         Model& setInputSize(int height, int width);
+         Model& setInputMean(const Scalar& mean);
+         Model& setInputScale(float scale);
+         Model& setInputCrop(bool crop);
+         Model& setInputSwapRB(bool swapRB);
+         std::pair<int, float> classify(InputArray frame);
+         std::vector<Mat> predict(const Mat& frame);
+
+     protected:
+         struct Impl;
+         Ptr<Impl> impl_;
+     };
+
+     class CV_EXPORTS_W_SIMPLE DetectionModel : public Model
```

Comment:
I think that we need to use `CV_EXPORTS_W` because `CV_EXPORTS_W_SIMPLE` is mostly used for some data structures but we have pretty complicated logic inside our new classes.